common/src/main/java/com/wynntils/handlers/bossbar/BossBarHandler.java

```diff
@@ -57,14 +55,18 @@ public void add(
                 boolean darkenScreen,
                 boolean playMusic,
                 boolean createWorldFog) {
-            Optional<Pair<TrackedBar, Matcher>> trackedBarOpt = knownBars.stream()
-                    .map(bar -> new Pair<>(bar, ComponentUtils.getCoded(name).getMatcher(bar.pattern)))
-                    .filter(pair -> pair.b().matches())
-                    .findFirst();
-            if (trackedBarOpt.isEmpty()) return;
 
-            TrackedBar trackedBar = trackedBarOpt.get().a();
-            Matcher matcher = trackedBarOpt.get().b();
+            TrackedBar trackedBar = null;
+            Matcher matcher = null;
+
+            for (TrackedBar bar : knownBars) {
+                matcher = StyledText.fromComponent(name).getMatcher(bar.pattern);
+                if (matcher.matches()) {
+                    trackedBar = bar;
+                    break;
+                }
+            }
+            if (trackedBar == null) return;
```

Comment:
Why was this changed?

Reply:
better readability

Reply:
Well, I think it is just as readable now as before. And now comes the magic sentence: "no unnecessary changes in porting PRs"